### ISSUE TITLE
Refactor Metal EC Backend

### ISSUE THREAD

**Issue by moven0831** [zkmopro/gpu-acceleration#20]
## Description
- PR #13 introduced a comprehensive refactoring and enhancement of the Metal MSM module, including an EC backend adapted from Tal and Koh's work on Zprize 2023, various arithmetic operations, benchmarking, and testing suites. The original PR is quite large and spans multiple distinct domains. To maintain clarity, improve review efficiency, and track incremental progress, we need to split the original PR into several smaller PRs.

## Goals
1. Sub PRs:   
   - **BigInt Operations:**  
     `BigInt` arithmetic functionalities (addition, subtraction for arbitrary limb size and tests).

   - **Field Operations:**  
     `Field` arithmetic functionalities (addition, subtraction above field).

   - **Montgomery Multiplication**  
     link to #6 

   - **Elliptic Curve Operations:**  
     `ECPoint` types definition and associated point addition/doubling operations.

2. Close the Original PR:
   Once the new PRs are created and reference this issue, the original PR will be closed in favor of the smaller, more manageable contributions.

**Acceptance Criteria:**
- Each new PR should be self-contained and properly documented.
- All new PRs should reference this issue to maintain proper traceability.
- The original PR should be closed only after these new PRs are created and are in review.

**References:**
- Original PR: #13
- Related Issue: #6
- https://github.com/z-prize/2023-entries/tree/main/prize-2-msm-wasm/webgpu-only/tal-derei-koh-wei-jie
- https://github.com/geometryxyz/msl-secp256k1
- https://github.com/geometers/multiprecision
- https://github.com/mitschabaude/montgomery#13-x-30-bit-multiplication

**Comment by moven0831** [comment]
- Within #31
- and sub issue #25

The EC algorithms for point addition (pADD), point mixed addition (pMADD), and point doubling (pDBL) successfully passed the correctness test using random inputs.
